restorer.go

```diff
@@ -73,19 +77,30 @@ func (res *Restorer) restoreTo(dst string, dir string, treeID backend.ID) error
 	return nil
 }
 
-func (res *Restorer) restoreNodeTo(node *Node, dir string, dst string) error {
+func (res *Restorer) restoreNodeTo(node *Node, dir string, dst string) (excluded bool, err error) {
+	debug.Log("Restorer.restoreNodeTo", "node %v, dir %v, dst %v", node.Name, dir, dst)
 	dstPath := filepath.Join(dst, dir, node.Name)
 
-	if res.Filter != nil && res.Filter(filepath.Join(dir, node.Name), dstPath, node) {
-		return nil
+	if res.SelectForRestore != nil {
+		debug.Log("Restorer.restoreNodeTo", "running select filter for %v", node.Name)
+
+		if !res.SelectForRestore(filepath.Join(dir, node.Name), dstPath, node) {
```

Comment:
I kinda feel like the filtering should be up to the caller and not happen in `restoreNodeTo`. Feels like this is not this method's responsibility.